pycbc/coordinates/space.py

```diff
@@ -0,0 +1,906 @@
+        # Earth is moving, when GW arrives at Earth center,
+        # time is t_geo, not t_ssb.
+        p = earth_position_ssb(t_geo)[0]
+        return t_geo - t_ssb - np.vdot(k, p) / c.value
+
+    return fsolve(equation, t_ssb)[0]
+
+
+def t_ssb_from_t_geo(t_geo, longitude_ssb, latitude_ssb):
+    """ Calculating the time when a GW signal arrives at the barycenter
+    of SSB, by using the time in geocentric frame and sky localization
+    in SSB frame.
+
+    Parameters
+    ----------
+    t_geo : float
+        The time when a GW signal arrives at the origin of geocentric frame.
+        In the unit of 's'.
+    longitude_ssb : float
+        The ecliptic longitude of a GW signal in SSB frame.
+        In the unit of 'radian'.
+    latitude_ssb : float
+        The ecliptic latitude of a GW signal in SSB frame.
+        In the unit of 'radian'.
+
+    Returns
+    -------
+    t_ssb : float
+        The time when a GW signal arrives at the origin of SSB frame.
+    """
+    k = localization_to_propagation_vector(longitude_ssb, latitude_ssb)
+    # Earth is moving, when GW arrives at Earth center,
+    # time is t_geo, not t_ssb.
+    p = earth_position_ssb(t_geo)[0]
+
+    def equation(t_ssb):
+        return t_geo - t_ssb - np.vdot(k, p) / c.value
+
+    return fsolve(equation, t_geo)[0]
+
+
+def ssb_to_geo(t_ssb, longitude_ssb, latitude_ssb, polarization_ssb,
+               use_astropy=True):
+    """ Converting the arrive time, the sky localization, and the polarization
+    from the SSB frame to the geocentric frame.
+
+    Parameters
+    ----------
+    t_ssb : float or numpy.array
+        The time when a GW signal arrives at the origin of SSB frame.
+        In the unit of 's'.
+    longitude_ssb : float or numpy.array
+        The ecliptic longitude of a GW signal in SSB frame.
+        In the unit of 'radian'.
+    latitude_ssb : float or numpy.array
+        The ecliptic latitude of a GW signal in SSB frame.
+        In the unit of 'radian'.
+    polarization_ssb : float or numpy.array
+        The polarization angle of a GW signal in SSB frame.
+        In the unit of 'radian'.
+    use_astropy : bool
+        Using Astropy to calculate the sky localization or not.
+        Default is True.
+
+    Returns
+    -------
+    (t_geo, longitude_geo, latitude_geo, polarization_geo) : tuple
+    t_geo : float or numpy.array
+        The time when a GW signal arrives at the origin of geocentric frame.
+        In the unit of 's'.
+    longitude_geo : float or numpy.array
+        The longitude of a GW signal in geocentric frame.
+        In the unit of 'radian'.
+    latitude_geo : float or numpy.array
+        The latitude of a GW signal in geocentric frame.
+        In the unit of 'radian'.
+    polarization_geo : float or numpy.array
+        The polarization angle of a GW signal in geocentric frame.
+        In the unit of 'radian'.
+    """
+    if isinstance(t_ssb, np.ndarray):
+        t_ssb_array = t_ssb.copy()
```

Comment:
Why do you have these copies? That is only  needed if some function you call modifies the vector in place. Otherwise, you don't need this. Most numpy operations  return a new array.